app/controllers/user_departments_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UserDepartmentsController < ApplicationController
+  before_action :find_department_id
+  before_action :logged_in_user,
+                except: %i(new create)
+  before_action :find_user_by_id, only: %i(create)
+  before_action ->{check_role? :admin}
+
+  def new
+    @user_department = @department.user_departments.build
+  end
+
+  def create
+    users_department = []
+    params[:user_ids].each do |user_id|
+      users_department << @department.user_departments.build(user_id: user_id)
+    end
+
+    ActiveRecord::Base.transaction do
+      users_department.map(&:save!)
+      redirect_to department_url(id: params[:department_id])
+    end
```

Comment:
transaction là gì? Sao lại phải sử dụng transaction ở đây

Reply:
- transaction là một tiến trình xử lý bao gồm nhiều tiến trình nhỏ bên trong, 1 transaction thành công là toàn bộ các tiến trình bên trong đều được xử lý thành công đồng thời, nếu 1 trong các tiến trình đó bị sai, transaction sẽ rollback về điểm xác định, mặc định là thời điểm trước khi thực thi transaction.
- sau khi dùng each để tạo user_derpartment tương ứng với mỗi phần tử trong mảng được truyền vào từ select, transaction giúp lưu lại toàn bộ users_department vào cơ sở dữ liệu